app/javascript/mastodon/storage/modifier.js

```diff
@@ -166,3 +179,27 @@ export function putStatuses(records) {
     .then(evictStatusesByRecords)
     .catch(printErrorIfAvailable);
 }
+
+export function freeStorage() {
+  return navigator.storage.estimate().then(({ quota, usage }) => {
+    if (usage + 8388608 < quota) {
```

Comment:
what does 8388608 represent here?

Reply:
I understood it as arbitrary padding to make sure the quota isn't even *close* to being exceeded, but I suppose for readability it would be better to extract this value into a constant with an appropriate name.